### PR TITLE
Fix ROOT-10895: accelerate DeepAutoLoadImpl()

### PR DESCRIPTION
**Pull request by Axel-Naumann**
In an LLVM-debug build that's reducing `time python3 -c 'import ROOT;x=ROOT.TPythia8()'` from 162s to 6.6s...

### CODE REVIEW

**Comment by Axel-Naumann** [comment]
Failures are pre-existing. @pcanal this is ready for review

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -6069,61 +6069,63 @@ Int_t TCling::ShallowAutoLoadImpl(const char *cls)
...
+            continue;
+         const char *subtypename = element->GetTypeName();
+         if (!TClassTable::GetDictNorm(subtypename)) {
+            // Failure to load a dictionary is not (quite) a failure load
+            // the top-level library.  If we return false here, then
+            // we would end up in a situation where the library and thus
+            // the dictionary is loaded for "cls" but the TClass is
+            // not created and/or marked as unavailable (in case where
+            // AutoLoad is called from TClass::GetClass).
+            DeepAutoLoadImpl(subtypename, visited, true /*normalized*/);
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx:6102]: The cast to `(void)` end up here because some compiler complained that the result was not used.

**Review by pcanal — APPROVED** [review]
The failure on fedora32 roottest.root.meta.ROOT-7462.roottest_root_meta_ROOT_7462_make is pre-existing but the test is directly affected by the change in this PR ... so hopefully we can repair the test (machine) before merging this in.

File: core/metacling/src/TCling.cxx (hunk 1 above)
Comment (Axel-Naumann) [reply core/metacling/src/TCling.cxx:6102]: OK but we don't seem to see those anymore?

**Pull request merged**